examples/econais/ec19d/sdk_example_kaa_client/src/kaa_client.c

```diff
@@ -0,0 +1,823 @@
+    }
+
+    ext_calculate_sha_hash(kaa_public_key, kaa_public_key_length, kaa_public_key_hash);
+    sndc_printf("SHA calculated\n");
+    return KAA_ERR_NONE;
+}
+
+/*
+ * Initializes Kaa log collector.
+ */
+kaa_error_t kaa_log_collector_init(kaa_client_t *kaa_client)
+{
+    KAA_RETURN_IF_NIL(kaa_client, KAA_ERR_BADPARAM)
+
+        kaa_error_t error_code = ext_unlimited_log_storage_create(
+                &kaa_client->log_storage_context,
+                kaa_client->kaa_context->logger);
+        if (error_code) {
+            KAA_LOG_ERROR(kaa_client->kaa_context->logger,
+                    error_code,
+                    "Failed to create log storage");
+            return error_code;
+        }
+
+        error_code = ext_log_upload_strategy_by_volume_create(&kaa_client->log_upload_strategy_context
+                                                            , kaa_client->kaa_context->channel_manager
+                                                            , kaa_client->kaa_context->bootstrap_manager);
+
+        if (error_code) {
+            KAA_LOG_ERROR(kaa_client->kaa_context->logger,
+                    error_code,
+                    "Failed to create log upload strategy");
+            return error_code;
+        }
+
+        error_code = ext_log_upload_strategy_by_volume_set_threshold_count(kaa_client->log_upload_strategy_context
+                                                                         , KAA_DEMO_UPLOAD_COUNT_THRESHOLD);
+        if (error_code) {
+            KAA_LOG_ERROR(kaa_client->kaa_context->logger,
+                    error_code,
+                    "Failed to create log upload strategy by volume set threshold count to %d",
+                    KAA_DEMO_UPLOAD_COUNT_THRESHOLD);
+            return error_code;
+        }
+
+        error_code = kaa_logging_init(kaa_client->kaa_context->log_collector
+                                    , kaa_client->log_storage_context
+                                    , kaa_client->log_upload_strategy_context);
+
+        if (error_code) {
+            KAA_LOG_ERROR(kaa_client->kaa_context->logger,
+                    error_code,
+                    "Failed to logging init");
+            return error_code;
+        }
+        KAA_LOG_INFO(kaa_client->kaa_context->logger,
+                KAA_ERR_NONE,
+                "Log collector init complete");
+        return error_code;
+
+}
+
+/*
+ * Example code for logging
+ */
+kaa_error_t kaa_client_log_record(kaa_client_t *kaa_client, const char *record)
+{
+    if (!kaa_client || !record) {
+        return KAA_ERR_BADPARAM;
+    }
+
+    kaa_logging_record_t * kaa_record = kaa_logging_record_create();
+    if (!kaa_record) {
+        KAA_LOG_ERROR(kaa_client->kaa_context->logger,
+                KAA_ERR_NOT_INITIALIZED,
+                "Failed to allocate log record");
+        return KAA_ERR_NOT_INITIALIZED;
+    }
+
+    kaa_record->body = kaa_string_move_create(record, NULL);
```

Comment:
According to the usage of this function, the right way is to use the kaa_string_copy_create version.